google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/data/v2/stub/readrows/ReadRowsRetryTest.java

```diff
@@ -124,6 +134,98 @@ public void immediateRetryTest() {
     Truth.assertThat(actualResults).containsExactly("k1", "r1", "r2").inOrder();
   }
 
+  public ApiException largeRowExceptionWithTrailers(String rowKey) {
+    ErrorInfo errorInfo =
+        ErrorInfo.newBuilder()
+            .setReason("LargeRowReadError")
+            .setDomain("bigtable.googleapis.com")
+            .putMetadata("rowKey", rowKey)
+            .build();
+
+    Any packedErrorInfo = Any.pack(errorInfo);
+    // ErrorDetails errorDetails = ErrorDetails.builder()
+    //     .setRawErrorMessages(Collections.singletonList(packedErrorInfo))
+    //     .build();
+
+    ErrorDetails errorDetails =
+        ErrorDetails.builder().setRawErrorMessages(ImmutableList.of(packedErrorInfo)).build();
+
+    Metadata trailers = new Metadata();
+    byte[] status =
+        com.google.rpc.Status.newBuilder().addDetails(Any.pack(errorInfo)).build().toByteArray();
+    trailers.put(ERROR_DETAILS_KEY, status);
+    return (ApiException)
+        (new UnavailableException(
+            new StatusRuntimeException(Status.FAILED_PRECONDITION, trailers),
+            GrpcStatusCode.of(Code.FAILED_PRECONDITION),
+            false,
+            errorDetails));
+  }
+
+  /**
+   * AdditionalTests - 1. only query for large rows - empty response? 2. 1st row is large row - then
+   * get other responses 3. last row is large row 4. multiple adhoc large rows 5. continous large
+   * rows
```

Comment:
I also don't think everything described in this comment is added. I think they're all valuable and we should add them.

Additionally, we should add a few more tests
1. Test row keys logic is correct. Query.create().rowKey(key1).rowKey(k2).rowKey(largeRow). 
2. Test the row limit behavior is still correct. 
3. Test reverse scan. `Query.create(..).reversed(true);` is set. `reversed(true)` will return rows in the reversed order, so instead of returning rows like "r1, r2, r3", it'll be "r3, r2, r1".